### PR TITLE
Improve error message for Hive tables where partition column is also in table

### PR DESCRIPTION
**Pull request by raghavsethi**

### CODE REVIEW

**Comment by haozhun** [comment]
@dain this is a short PR fixing an uncategorized error we observed in warehouse. It looks good to me. Please take a second look.

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java [hunk 1]
```diff
@@ -1379,8 +1380,14 @@ private static HiveStorageFormat extractHiveStorageFormat(Table table)
 
     private static Function<HiveColumnHandle, ColumnMetadata> columnMetadataGetter(Table table, TypeManager typeManager)
     {
+        Set<String> partitionKeyNames = table.getPartitionKeys().stream().map(FieldSchema::getName).collect(Collectors.toSet());
+        List<FieldSchema> tableColumns = table.getSd().getCols();
+        if (tableColumns.stream().anyMatch(field -> partitionKeyNames.contains(field.getName()))) {
+            throw new PrestoException(HIVE_INVALID_METADATA, "Partition column cannot also be a table column");
```
Comment (martint) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java]: This is a weird message. From Presto's (and Presto users) point of view, partition columns *are* table columns.

What's the issue we're trying to deal with here? What fails if a column name appears in both?

Comment (raghavsethi) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java]: The issue is that we build a map immediately afterwards, which fails when we have two entries with the same key. @haozhun said that partition columns should not appear in the table, which is why I chose this error message.

Comment (martint) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java]: The question I have is what's the harm in having partition columns appear in the list of table columns (coming from the metastore). Is that indicative of corrupt metadata or is it an expected state that normally doesn't happen? Would it be safe to just ignore them? What does Hive do in this case?

So, the two options I see are:
* Change the code to be able to handle this scenario
* Fail with a clearer error message. Something like: Hive metadata for table xxx is corrupt. Partition keys are also listed in the storage descriptor"

**Comment by haozhun** [comment]
@raghavsethi Maybe we can change the check so that a failure message is thrown whenever duplicate column name exists (2 partition columns with same column name, 2 non-partition columns with same column name, or 1 partition column and 1 non-partition column name has the same column name). This way, we can just say duplicate column name exists.

@martint I'm not convinced that presto users generally agree that partition columns are table columns. After all, many of them come from Hive background. Like when we released 0.123, the first user that tried out our new feature of creating a table wrote:
```
CREATE TABLE IF NOT EXISTS test_presto_part (
    dummy BIGINT
) WITH (partitioned_by = ARRAY['ds']);
```

**Comment by martint** [comment]
> Like when we released 0.123, the first user that tried out our new feature of creating a table wrote:

Yes, and they had to fix it because partition columns must be table columns. A lot of things are weird in Hive, but user familiarity doesn't mean we should do the same in Presto. Presto is not Hive :)

**Comment by raghavsethi** [comment]
@haozhun: Martin asked me to look at how Hive behaves with this table, and it fails with this message:
```
FAILED: SemanticException Partition column name host conflicts with table columns.
Caused by: HiveException: Partition column name host conflicts with table columns.
```
Given that this is an error message specific to Hive, I think the new error message is the best best. We can add an additional check to verify that no two columns (partitioned or not) have the same name.

**Comment by martint** [comment]
Looks good

**Pull request merged**